src/analysis/processing/qgsalgorithmrastercalculator.cpp

```diff
@@ -0,0 +1,192 @@
+/***************************************************************************
+                         qgsalgorithmrastercalculator.cpp
+                         ---------------------
+    begin                : July 2023
+    copyright            : (C) 2023 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmrastercalculator.h"
+#include "qgsrasterfilewriter.h"
+#include "qgsrastercalculator.h"
+
+///@cond PRIVATE
+
+QString QgsRasterCalculatorAlgorithm::name() const
+{
+  return QStringLiteral( "rastercalc" );
+}
+
+QString QgsRasterCalculatorAlgorithm::displayName() const
+{
+  return QObject::tr( "Raster calculator" );
+}
+
+QStringList QgsRasterCalculatorAlgorithm::tags() const
+{
+  return QObject::tr( "raster,calculator" ).split( ',' );
+}
+
+QString QgsRasterCalculatorAlgorithm::group() const
+{
+  return QObject::tr( "Raster analysis" );
+}
+
+QString QgsRasterCalculatorAlgorithm::groupId() const
+{
+  return QStringLiteral( "rasteranalysis" );
+}
+
+QString QgsRasterCalculatorAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Performing algebraic operations using raster layers." );
+}
+
+QgsRasterCalculatorAlgorithm *QgsRasterCalculatorAlgorithm::createInstance() const
+{
+  return new QgsRasterCalculatorAlgorithm();
+}
+
+void QgsRasterCalculatorAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterMultipleLayers( QStringLiteral( "INPUT" ), QObject::tr( "Input layers" ), QgsProcessing::SourceType::TypeRaster ) );
+  addParameter( new QgsProcessingParameterExpression( QStringLiteral( "EXPRESSION" ), QObject::tr( "Expression" ), QVariant(),  QStringLiteral( "INPUT" ), false, Qgis::ExpressionType::RasterCalculator ) );
+  addParameter( new QgsProcessingParameterExtent( QStringLiteral( "EXTENT" ), QObject::tr( "Output extent" ), QVariant(), true ) );
```

Comment:
For these parameters I'd suggest adding setHelp with descriptions like "Extent of the output layer. If not specified, the extent will be the overall extent of all input layers".  "... if no CRS is specified, then the CRS of the first input layer will be used", etc.

Reply:
Done.